packages/cli/src/modules/chat-hub/chat-hub-workflow.service.ts

```diff
@@ -698,6 +694,111 @@ When you need to reference “now”, use this date and time.`;
 		};
 	}
 
+	private async buildMessageValuesWithAttachments(
+		provider: ChatHubProvider,
+		history: ChatHubMessage[],
+	): Promise<MessageRecord[]> {
+		// Gemini has 20MB limit, the value should also be what n8n instance can safely handle
+		const maxTotalPayloadSize = 20 * 1024 * 1024 * 0.9;
+
+		const typeMap: Record<string, MessageRecord['type']> = {
+			human: 'user',
+			ai: 'ai',
+			system: 'system',
+		};
+
+		const messageValues: MessageRecord[] = [];
+		let currentTotalSize = 0;
+
+		for (const message of history.slice().reverse()) {
```

Comment:
Traversing messages from last to prioritize newer attachments (in case the limit is reached)